### PR TITLE
node-glob: use buildNpmPackage

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
See #229475

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/glob/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+}:
+
+buildNpmPackage rec {
+  pname = "glob";
```
Comment (fgaz) [inline pkgs/tools/misc/glob/default.nix:7]: we might want to name it node-glob [like most distros](https://repology.org/project/node:glob/versions)

File: pkgs/tools/misc/glob/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  npmDepsHash = "sha256-78oODw+CBCk5JRJbDqLqVmzTVImP7Z7o6jRIimDxZDQ=";
+
+  dontNpmBuild = true;
+
+  meta = {
+    changelog = "https://github.com/isaacs/node-glob/blob/${src.rev}/changelog.md";
+    description = "The most correct and second fastest glob implementation in JavaScript";
+    homepage = "https://github.com/isaacs/node-glob";
+    license = lib.licenses.isc;
+    mainProgram = "glob";
```
Comment (fgaz) [inline pkgs/tools/misc/glob/default.nix:26]: if we name the package "glob", this line is unnecessary

File: pkgs/tools/misc/glob/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    hash = "sha256-oLlNhQOnu/hlKjNWa5vjqslz1EarZJOpUEXUB+vGQvc=";
+  };
+
+  npmDepsHash = "sha256-78oODw+CBCk5JRJbDqLqVmzTVImP7Z7o6jRIimDxZDQ=";
+
+  dontNpmBuild = true;
+
+  meta = {
+    changelog = "https://github.com/isaacs/node-glob/blob/${src.rev}/changelog.md";
+    description = "The most correct and second fastest glob implementation in JavaScript";
```
Comment (fgaz) [inline pkgs/tools/misc/glob/default.nix:23]: This reads more like a slogan than a description

**Review by fgaz — APPROVED** [review]
Looks good but needs a rebase. I left a bunch of suggestions/nitpicks

---

3 packages built:
glob openmoji-black openmoji-color

Result of `nixpkgs-review pr 250077` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>glob</li>
    <li>openmoji-black</li>
    <li>openmoji-color</li>
  </ul>
</details>

File: pkgs/tools/misc/glob/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/tools/misc/glob/default.nix:26]: not anymore: https://github.com/NixOS/nixpkgs/pull/246386

File: pkgs/tools/misc/glob/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/tools/misc/glob/default.nix]: I went with the most common description on Repology.

**Review by Lord-Valen — COMMENTED** [review]
Just needs a maintainer.

File: pkgs/tools/misc/glob/default.nix (hunk 2 above)
Comment (fgaz) [reply pkgs/tools/misc/glob/default.nix]: Wow, looks like I have a few patches to submit then

**Comment by fgaz** [comment]
@dotlambda would you like to maintain the package?

**Comment by dotlambda** [comment]
> @dotlambda would you like to maintain the package?

No.

**Review by fgaz — APPROVED** [review]
Haha well I tried. Let's merge then, it's not like this had a maintainer before, and node-packages.nix is prone to conflicts

**Pull request merged**